src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs

```diff
@@ -216,10 +216,11 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
                 node = ctx.CreateNode(opType, ctx.GetVariableName(outColumnNames[1]), binarizerOutput, ctx.GetNodeName(opType));
                 node.AddAttribute("threshold", 0.0);
             }
+
             opType = "Cast";
             node = ctx.CreateNode(opType, binarizerOutput, ctx.GetVariableName(outColumnNames[0]), ctx.GetNodeName(opType), "");
-            var t = InternalDataKindExtensions.ToInternalDataKind(DataKind.Boolean).ToType();
-            node.AddAttribute("to", t);
+            var predictedLabelCol = OutputSchema.GetColumnOrNull(outColumnNames[0]);
+            node.AddAttribute("to", predictedLabelCol.HasValue ? predictedLabelCol.Value.Type.RawType : typeof(bool));
```

Comment:
>predictedLabelCol.HasValue [](start = 36, length = 26)

Doesn't it always have a value?
Or, if it doesn't - should we be adding the ONNX node at all?

Reply:
I have added an Assert to capture that and fixed the next line.

---
In reply to: [376929923](https://github.com/dotnet/machinelearning/pull/4818#discussion_r376929923) [](ancestors = 376929923)